docs/DeveloperGuide.md

```diff
@@ -162,40 +162,162 @@ Classes used by multiple components are in the `seedu.realodex.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
-``### Filter by Name feature
+### Overall Filter feature
 
 #### Implementation
+This feature enables users to filter and view persons in Realodex based on specified criteria such as name, remark, tag, birthday, and housing type.
+The key components involved are:
```

Comment:
might want to follow our standard order 
name, tag, housing type, remark, birthday iirc

Reply:
sounds good, thanks!